include/swift/FrontendTool/FrontendTool.h

```diff
@@ -42,6 +42,22 @@ class FrontendObserver {
 
   /// The frontend has configured the compiler instance.
   virtual void configuredCompiler(CompilerInstance &instance);
+
+  /// The frontend has performed semantic analysis.
+  virtual void performedSemanticAnalysis(CompilerInstance &instance);
+
+  /// The frontend has performed basic SIL generation.
+  /// SIL diagnostic passes have not yet been applied.
+  virtual void performedSILGeneration(SILModule &module);
+
+  /// The frontend has executed the SIL optimization and diagnostics pipelines.
+  virtual void performedSILProcessing(SILModule &module);
+
+  /// The frontend is about to run the program as an immediate script.
+  virtual void aboutToRunImmediately(CompilerInstance &instance);
```

Comment:
Can we leave this one out for now? It seems a little different from the others, and we don't know of any clients.